### PR TITLE
JDK9 build compatibility fixes

### PR DESCRIPTION
**Pull request by psiroky**
These changes should currently be enough for the simple `mvn clean install` build. We will likely need further fixes for tests - but that can be done separately.

### CODE REVIEW

**Comment by psiroky** [comment]
Jenkins execute full downstream build

**Comment by psiroky** [comment]
Jenkins execute full downstream build

**Comment by psiroky** [comment]
I've proposed a fix for revapi: https://github.com/revapi/revapi/pull/99. We will need to wait for a new release unfortunately :(

**Comment by psiroky** [comment]
Jenkins execute full downstream build

**Comment by psiroky** [comment]
Jenkins execute full downstream build

File: pom.xml [hunk 1]
```diff
@@ -1584,28 +1586,22 @@
       </activation>
       <properties>
         <!-- There is no public ECJ release (yet) which supports Java 9, so fallback to the native compiler -->
-        <!-- TODO remove the line below (and fix the surefire+failsafe config) after resolving https://issues.jboss.org/browse/DROOLS-1169 -->
-        <drools.dialect.java.compiler>NATIVE</drools.dialect.java.compiler>
```
Comment (ge0ffrey) [inline pom.xml]: Are you sure this line can be removed?
It's for running the tests, not compation and the related ECJ issue DROOLS-1169 isn't resolved yet.

File: pom.xml [hunk 2]
```diff
@@ -174,6 +174,8 @@
          IMPORTANT: always explain the reason for overriding the plugin version! -->
     <!-- Surefire version 2.19.1 coming from parent is buggy and kie-osgi tests in droolsjbpm-integration are failing because of that. -->
     <version.surefire.plugin>2.18.1</version.surefire.plugin>
+    <version.enforcer.plugin>3.0.0-M1</version.enforcer.plugin> <!-- Needed to support JDK 9 -->
+    <version.javadoc.plugin>3.0.0-M1</version.javadoc.plugin> <!-- Needed to support JDK 9 -->
```
Comment (ge0ffrey) [inline pom.xml]: Can we add something in the comment as to when will we be able to remove these overrides?

We depend on ipbom 7.0 which extends jboss parent 22.
but ipbom 8.0 already extends jboss parent 25, which might already be JDK 9 compliant (as WildFly uses it too)

**Comment by ge0ffrey** [comment]
Good to merge, only soft suggestions

File: pom.xml (hunk 1 above)
Comment (psiroky) [reply pom.xml]: Yes, I just tried it locally and all works as expected with ECJ 4.4.2.

File: pom.xml (hunk 2 above)
Comment (psiroky) [reply pom.xml]: I will add a comment, but we should find a better way to detect these. E.g. have a script that checks which property values are exactly the same as in parent, so can be safely removed. But that will of course require some time to implement.

**Comment by psiroky** [comment]
Jenkins execute full downstream build

**Pull request merged**